### PR TITLE
PyTorch migration: Add RankNet estimators

### PR DESCRIPTION
**Pull request by timokau**
## Description

This adds pytorch versions of the CmpNet estimators.

## Motivation and Context

Part of the ongoing pytorch migration.

## How Has This Been Tested?

I have added the RankNet estimators to the existing tests for discrete choice, general choice and object ranking. I have checked the changes with our set of linters and the test suite.

## Does this close/impact existing issues?

Related to #125.

## Types of changes

This depends on what you consider as the base. It is a new feature for the pytorch migration branch, but a breaking change on master.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: csrank/choicefunction/ranknet_choice.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    n_units : int
+        The number of units per hidden layer that should be used in the utility
+        module.
+
+    activation : torch activation function (class)
+        The activation function that should be used for each layer of the
+        comparative network.
+
+    criterion : torch criterion (class)
+        The criterion that is used to evaluate and optimize the module.
```
Comment (kiudee) [inline csrank/choicefunction/ranknet_choice.py:29]: Very minor remark: The order of the parameters in the docstring and the __init__ signature is swapped.

**Review by kiudee — APPROVED** [review]
I only found a very minor docstring issue (see above). Otherwise it looks good to me.

File: csrank/choicefunction/ranknet_choice.py (hunk 1 above)
Comment (timokau) [reply csrank/choicefunction/ranknet_choice.py:29]: Good catch. Details matter. Changed :)

**Comment by timokau** [comment]
I have pushed an update. Please take another look.

**Review by kiudee — APPROVED** [review]
Looks good to me.

**Comment by timokau** [comment]
Thanks for the review :)

**Pull request merged**